src/renderer/atlas/AtlasEngine.r.cpp

```diff
@@ -749,102 +810,19 @@ void AtlasEngine::CachedGlyphLayout::undoScaling(ID2D1RenderTarget* d2dRenderTar
 
 void AtlasEngine::_d2dPresent()
 {
-    const til::rect fullRect{ 0, 0, _r.cellCount.x, _r.cellCount.y };
-
-    // A change in the selection or background color (etc.) forces a full redraw.
-    if (WI_IsFlagSet(_r.invalidations, RenderInvalidations::ConstBuffer))
+    if (!_r.d2dRenderTarget)
     {
-        _r.dirtyRect = fullRect;
-    }
-
-    if (!_r.dirtyRect)
-    {
-        return;
-    }
-
-    auto dirtyRectInPx = _r.dirtyRect;
-    dirtyRectInPx.left *= _r.fontMetrics.cellSize.x;
-    dirtyRectInPx.top *= _r.fontMetrics.cellSize.y;
-    dirtyRectInPx.right *= _r.fontMetrics.cellSize.x;
-    dirtyRectInPx.bottom *= _r.fontMetrics.cellSize.y;
-
-    if (const auto intersection = _r.previousDirtyRectInPx & dirtyRectInPx)
-    {
-        wil::com_ptr<ID3D11Resource> backBuffer;
-        wil::com_ptr<ID3D11Resource> frontBuffer;
-        THROW_IF_FAILED(_r.swapChain->GetBuffer(0, __uuidof(backBuffer), backBuffer.put_void()));
-        THROW_IF_FAILED(_r.swapChain->GetBuffer(1, __uuidof(frontBuffer), frontBuffer.put_void()));
-
-        D3D11_BOX intersectBox;
-        intersectBox.left = intersection.left;
-        intersectBox.top = intersection.top;
-        intersectBox.front = 0;
-        intersectBox.right = intersection.right;
-        intersectBox.bottom = intersection.bottom;
-        intersectBox.back = 1;
-        _r.deviceContext->CopySubresourceRegion1(backBuffer.get(), 0, intersection.left, intersection.top, 0, frontBuffer.get(), 0, &intersectBox, 0);

```

Comment:
we don't need to do this manual copy anymore? Is this cause of the `DXGI_SWAP_EFFECT_FLIP_SEQUENTIAL` change?

Reply:
No I think this was never necessary and only fixed a visual glitch I had before I fixed other bugs in the D2D drawing code.